tracing/src/main/java/com/palantir/remoting3/tracing/CloseableTracer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.remoting3.tracing;
+
+import com.palantir.remoting.api.tracing.SpanType;
+
+/**
+ * Wraps the {@link Tracer} methods in a closeable resource to enable the usage of the try-with-resources pattern.
+ *
+ * Usage:
+ *   try (CloseableTracer trace = CloseableTracer.start("traceName")) {
+ *       [...]
+ *   }
```

Comment:
Hey @raiju any chance we could also add one sentence of guidance so people know roughly when they should use this? The main thing I want to convey is that http-remoting calls are already traced, so people don't need to wrap every single call they ever do in this!

Reply:
Is this class that different from https://github.com/palantir/http-remoting/blob/29b83d50358d17e8f57eca62b2790ec6e08ecbf6/tracing/src/main/java/com/palantir/remoting3/tracing/Tracer.java#L42 to warrant this?